web-basics-browser-security-model/README.md

```diff
@@ -0,0 +1,403 @@
+
+# Talking to web sites
+
+Let's go through the basics of how HTTP requests are made, using telnet to form requests. First, let's connect to the vulnerable web server:
+
+```
+user@hostname~#: telnet 141.85.224.157 80
+```
+
+Now let's issue a simple GET. The request is composed of:
+
+- GET <path> <http-version>
+- followed by other header contents
+- followed by an additional newline, indicating the end of the request.
+
+_Please note the above bullet points and the fact that you need to provide an additional newline to indicate the end of the request._
+
+```
+GET / HTTP/1.0
+```
+
+```
+HEAD / HTTP/1.0
+```
+
+```
+POST / HTTP/1.0
+```
+
+The server's response contains:
+
+- A status code (200 OK in our case)
+- Date information and information about the server
+- Encoding and other info about the data, i.e. it's MIME-type
+- The length of the data
+- The actual data
+
+# Tools
+
+## Postman
+
+Postman is a tool through which you can send HTTP requests. It can automate tests and integrate them into CI/CD pipeline. You can simulate endpoints to test without having to set up a backend. It can also monitor certain endpoints to check their health and performance. [[6]](https://www.postman.com/)
+
+## HTTPie
+
+HTTPie is also used to send HTTP requests. It can be used in the terminal, instead of utilities such as **curl** and **wget**. It has an easy syntax, comes with highlighting and JSON assets. [[7]](https://httpie.io/)
+
+## Hoppscotch
+
+Hoppscotch is a tool that you can use from your browser or as a browser extension to quickly test certain endpoints. It also assetss MQTT or GraphQL requests. [[8]](https://hoppscotch.io/)
+
+# Further Reading
+
+- [1] https://skorks.com/2010/05/what-every-developer-should-know-about-urls/
+- [2] https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
+- [3] https://owasp.org/www-project-top-ten/
+- [4] https://github.com/denimgroup/denimgroup-vulnerability-examples
+- [5] https://portswigger.net/web-security/cors/same-origin-policy
+- [6] https://www.postman.com/
+- [7] https://httpie.io/
+- [8] https://hoppscotch.io/
+
+# Challenges
+
+**1.** The below image represents a snippet with DevTools containing information about a web application. What can you discover in the next image ? Is there any useful information from a security point of view ? Write the answer to the instructor.
+
+![inspect-element](https://drive.google.com/uc?export=view&id=17xxCWiNyMQtCec5tRYU-lNOk4vwr9EUV)
```

Comment:
Can we also upload this image here, in `assets/` or in `activities/inspect-element/public/`? And not depend on external websites anymore.